src/js/https_client.js

```diff
@@ -0,0 +1,185 @@
+
+  if (options.auth) {
+    var headerString = 'Authorization: Basic ' + toBase64(options.auth);
+    httpsNative.addHeader(headerString, this);
+  }
+  if (options.headers) {
+    var keys = Object.keys(options.headers);
+    for (var i = 0, l = keys.length; i < l; i++) {
+      var key = keys[i];
+      httpsNative.addHeader(key + ': ' + options.headers[key], this);
+    }
+  }
+  httpsNative.sendRequest(this);
+}
+
+util.inherits(ClientRequest, stream.Writable);
+
+// Concrete stream overriding the empty underlying _write method.
+ClientRequest.prototype._write = function(chunk, callback, onwrite) {
+  httpsNative._write(this, chunk.toString(), callback, onwrite);
+};
+
+ClientRequest.prototype.headersComplete = function() {
+  var self = this;
+  self.emit('response', self._incoming);
+  var isHeadResponse = (self.method == 'HEAD');
+  return isHeadResponse;
+};
+
+ClientRequest.prototype.onError = function(ret) {
+  this.emit('error', ret);
+};
+
+ClientRequest.prototype.onFinish = function() {
+  httpsNative.finishRequest(this);
+};
+
+ClientRequest.prototype.setTimeout = function(ms, cb) {
+  this._incoming.setTimeout(ms, cb);
+};
+
+ClientRequest.prototype.abort = function(doNotEmit) {
+  if (!this.aborted) {
+    httpsNative.abort(this);
+    var date = new Date();
+    this.aborted = date.getTime();
+
+    if (this._incoming.parser) {
+      this._incoming.parser.finish();
+      this._incoming.parser = null;
+    }
+
+    if (!doNotEmit) {
+      this.emit('abort');
+    }
+  }
+};
+
+exports.ClientRequest = ClientRequest;
+
+//Function for encoding options.auth
+function utf8Encode(string) {
+  string = string.replace(/\r\n/g, '\n');
+  var utftext = '';
+
+  for (var n = 0; n < string.length; n++) {
+
+    var c = string.charCodeAt(n);
+
+    if (c < 128) {
+      utftext += String.fromCharCode(c);
+    } else if ((c > 127) && (c < 2048)) {
+      utftext += String.fromCharCode((c >> 6) | 192);
+      utftext += String.fromCharCode((c & 63) | 128);
+    } else {
+      utftext += String.fromCharCode((c >> 12) | 224);
+      utftext += String.fromCharCode(((c >> 6) & 63) | 128);
+      utftext += String.fromCharCode((c & 63) | 128);
+    }
+
+  }
+
+  return utftext;
+}
+
+function toBase64(input) {
+  var output = '';
+  var chr1, chr2, chr3, enc1, enc2, enc3, enc4;
+  var i = 0;
+
+  input = utf8Encode(input);
+  var _keyStr = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz' +
+    '0123456789+/=';
+
+  while (i < input.length) {
+
+    chr1 = input.charCodeAt(i++);
+    chr2 = input.charCodeAt(i++);
+    chr3 = input.charCodeAt(i++);
+
+    enc1 = chr1 >> 2;
+    enc2 = ((chr1 & 3) << 4) | (chr2 >> 4);
+    enc3 = ((chr2 & 15) << 2) | (chr3 >> 6);
+    enc4 = chr3 & 63;
+
+    if (isNaN(chr2)) {
+      enc3 = enc4 = 64;
+    } else if (isNaN(chr3)) {
+      enc4 = 64;
+    }
+
+    output = output +
+    _keyStr.charAt(enc1) + _keyStr.charAt(enc2) +
+    _keyStr.charAt(enc3) + _keyStr.charAt(enc4);
+
+  }
```

Comment:
No they are not. The are the leftover of since deleted debug statements, and I had (mistakenly) expected `clang-format` will just take care of such issues. I will remove such extra lines.